enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,621 @@
+`192.168.122.0/24` and deny the rest, even internal traffic.
+
+In order for this to work, and due to MicroShift's CNI, ovnk, we need to make
+further adjustments to services.
+
+The router is exposed through the use of a LoadBalancer service. This service
+will create iptables rules matching on the node IP:
+```bash
+# iptables -s nat -L
+…
+Chain OVN-KUBE-EXTERNALIP (2 references)
+target     prot opt source               destination
+DNAT       tcp  --  anywhere             microshift-dev       tcp dpt:https to:10.43.98.31:443
+DNAT       tcp  --  anywhere             microshift-dev       tcp dpt:http to:10.43.98.31:80
+```
+Traffic towards the router will be DNATted to the node ip, and then reach the
+service. All services default to use `externalTrafficPolicy: Cluster`, which
+will SNAT the client source IP to match that one of the node. This will get
+further NATing as described [here](https://github.com/openshift/microshift/blob/main/docs/contributor/network/ovn_kubernetes_traffic_flows.md#host-to-clusterservice).
+By the time the traffic reaches the NetworkPolicy checks the source IP has
+been replaced with `100.64.0.0/16`, which will not produce a match and traffic
+will get rejected.
+
+To overcome this, all LoadBalancer services in MicroShift that are subject to
+get NetworkPolicy resources to control external traffic must set
+`externalTrafficPolicy: Local`. This setting will not SNAT packets and the
+source IP is preserved until NetworkPolicy checks run.
+
+This setting, however, comes at the expense an even load when having more than
+one node, as traffic will not be sent to another node on the same port. This
+is not an issue in single node, which is MicroShift's use case and design
+principle.
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for the router prior to
+   MicroShift's start.
+2. After MicroShift started, the system will ingest the configuration and setup
+   everything according to it.
+3. The router will enabled/disabled, be exposed on the specified ports, on the
+   specified IPs, allowing only specified IP addresses, all according to the
+   cluster admin provided configuration. This includes the status in the
+   LoadBalancer service type for the router.
+
+### API Extensions
+None
+
+### Implementation Details
+The default router is composed of a bunch of assets part of the MicroShift
+binary. These assets come from the rebase, copied from the original router in
+[cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator).
+There is already a [LoadBalancer service](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/manifests/assets/router/service-cloud.yaml)
+included, which MicroShift does not copy yet. Depending on the options
+configured in MicroShift, this resource will need further customization done
+by MicroShift's start process.
```

Comment:
It would be good to have more info here about how the configuration options map to settings in the various manifests.

Reply:
Described how each configuration option changes manifests